snapcraft/internal/lifecycle.py

```diff
@@ -39,33 +39,84 @@
 from snapcraft.internal.cache import SnapCache
 from snapcraft.internal.indicators import is_dumb_terminal
 from snapcraft.internal.project_loader import replace_attr
+from snapcraft.internal.sources import _get_source_type_from_uri
 
 
 logger = logging.getLogger(__name__)
 
 
-_TEMPLATE_YAML = """name: my-snap-name # you probably want to 'snapcraft register <name>'
-version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
-summary: Single-line elevator pitch for your amazing snap # 79 char long summary
+_TEMPLATE_YAML = """name: {name} # you probably want to 'snapcraft register <name>'
+version: '{version}' # just for humans, typically '1.2+git' or '1.3.2'
+summary: {summary} # 79 char long summary
 description: |
-  This is my-snap's description. You have a paragraph or two to tell the
-  most important story about your snap. Keep it under 100 words though,
-  we live in tweetspace and your description wants to look good in the snap
-  store.
+  {description}
 
 grade: devel # must be 'stable' to release into candidate/stable channels
 confinement: devmode # use 'strict' once you have the right plugs and slots
 
+apps:
+  {name}:
+    command: {command} # Command to run this app
+    daemon: {daemon}
+    plugs: {plugs}
+    
 parts:
-  my-part:
+  {name}:
     # See 'snapcraft plugins'
-    plugin: nil
+    plugin: {plugin}
+    source: {source}
+    source-type: {source-type}
 """  # noqa, lines too long.
 
 _STEPS_TO_AUTOMATICALLY_CLEAN_IF_DIRTY = {'stage', 'prime'}
 
-
-def init():
+_PLUGIN_HINT_FILES = {
```

Comment:
No, it's just a start, we should add more items to this list to cover more plugins